cartoframes/dataset.py

```diff
@@ -116,9 +116,14 @@ def upload(self, with_lonlat=None, if_exists='fail', table_name=None, schema=Non
         return self
 
     def download(self, limit=None, decode_geom=False, retry_times=DEFAULT_RETRY_TIMES):
+        if self.cc is None or (self.table_name is None and self.query is None):
+            raise CartoException('You should provide a context and a table_name or query to download data.')
```

Comment:
Is this more of `ValueError` instead since it's not an error that occurs on CARTO's platform?

Reply:
:+1: